tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,511 @@
+
+### Why This Matters
+
+Without a unified tool layer:
+
+- Contributors spend time re-learning differing per-language conventions.
+- CI or release pipelines fail late due to missing metadata or mismatched versions.
+- Agents cannot reliably assist because state transitions ("now bump version", "now update changelog") aren't formalized.
+- Management-plane vs data-plane discrepancies introduce confusion (e.g., when to bump versions or generate changelog entries).
+
+Formalizing Stage 6 as cohesive tools improves reliability, supports automation-first workflows, and prepares for future LLM-assisted tasks.
+
+---
+
+## Goals and Exceptions/Limitations
+
+### Goals
+
+- [ ] Provide three composable, idempotent tools exposed via CLI and MCP agent: Changelog Update, Version Update, Metadata Update.
+- [ ] Normalize execution order and expected outputs across all supported languages.
+- [ ] Encode management-plane vs data-plane behavioral differences explicitly (especially version & changelog logic).
+- [ ] Emit structured JSON results (status + message + next steps hints) enabling orchestration by AI agents and pipelines.
+- [ ] Support configuration-driven per-language scripting (path or inline) without recompiling the tool.
+
+### Exceptions
+
+| Exception | Description | Impact | Workaround |
+|---|---|---|---|
+| Data‑Plane Deferred Version Bump | Data‑plane packages typically defer version adjustment and release‑date updates until the release‑preparedness phase. | The `update-version` tool may be a no‑op for data‑plane flows during Stage 6. | Keep version changes manually for milestone 1. |
+| Changelog Auto‑Generation (Data‑Plane) | Automatic changelog generation is not performed for data‑plane packages; manual editing is expected. | The `update-changelog-content` tool should not auto‑generate entries for data‑plane packages and should instead surface guidance. | Return a `noop` result with `next_steps` prompting manual edits. Consider future LLM integration to automate this post milestone 1. |
+
+### Language-Specific Limitations
+
+| Language   | Limitation | Impact | Workaround |
+|------------|------------|--------|------------|
+| JavaScript | Code formatting isn't hooked up to the emitter | Noisy diffs or linter failures | Run code formatting step in the `update-metadata` tool |
+
+---
+
+## Design Proposal
+
+### Overview
+
+Implement three individual CLI + MCP tools. Each tool:
+
+1. Resolves package path & language.
+2. Loads configuration for the language repository.
+3. If a script is configured, executes the script.
+4. Otherwise falls back the CLI's built-in implementation for that language.
+5. Aggregates structured result JSON with a `result`, human-readable `message`, and optional `next_steps` hint guiding subsequent tool invocation.
+
+Note: see the [repo-tooling-contract](./99-repo-tooling-contract.md) spec for more details.
```

Comment:
Let's not reference this spec yet until we get it finalized.

Instead based on your current implementation at https://github.com/Azure/azure-sdk-tools/pull/12761 you are using the existing swagger_to_sdk_config.json configuration so please add that here with an example. That will also help answer some of the questions in your implementation PR.

Reply:
Removed this link and added the configuration section and a sample.